compiler/src/prog8/compiler/target/cpu6502/codegen/AsmGen.kt

```diff
@@ -1308,18 +1309,30 @@ $repeatLabel    lda  $counterVar
         }
     }
 
+    /**
+     * TODO: %asminclude and %asmbinary should be done earlier than code gen (-> put content into AST)
+     */
     private fun translate(stmt: Directive) {
         when(stmt.directive) {
             "%asminclude" -> {
-                val sourcecode = loadAsmIncludeFile(stmt.args[0].str!!, stmt.definingModule().source)
+                val includedName = stmt.args[0].str!!
+                val sourcePath = stmt.definingModule().source // FIXME: %asminclude inside non-library, non-filesystem module
+                val sourcecode = loadAsmIncludeFile(includedName, sourcePath)
                 assemblyLines.add(sourcecode.trimEnd().trimStart('\n'))
             }
             "%asmbinary" -> {
+                val includedName = stmt.args[0].str!!
                 val offset = if(stmt.args.size>1) ", ${stmt.args[1].int}" else ""
                 val length = if(stmt.args.size>2) ", ${stmt.args[2].int}" else ""
-                val includedSourcePath = stmt.definingModule().source.resolveSibling(stmt.args[0].str)
-                val relPath = Paths.get("").relativize(includedSourcePath)
-                out("  .binary \"$relPath\" $offset $length")
+                val sourcePath = stmt.definingModule().source // FIXME: %asmbinary inside non-library, non-filesystem module
+                val includedPath = sourcePath.resolveSibling(includedName)
+
+                val pathForAssembler = outputDir // #54: 64tass needs the path *relative to the .asm file*
+                    .absolute() // avoid IllegalArgumentExc due to non-absolute path .relativize(absolute path)
+                    .relativize(includedPath)
```

Comment:
I'm not sure what's going on here but when I try out this merged PR locally, the compiler crashes here with the error below, if we call it with "-target build".    Note that these things work fine in your testing123 source branch.

`Exception in thread "main" java.lang.IllegalArgumentException: 'other' is different type of Path
        at java.base/sun.nio.fs.UnixPath.relativize(UnixPath.java:429)
        at java.base/sun.nio.fs.UnixPath.relativize(UnixPath.java:43)
        at prog8.compiler.target.cpu6502.codegen.AsmGen.translate(AsmGen.kt:1332)
        at prog8.compiler.target.cpu6502.codegen.AsmGen.translate$compiler(AsmGen.kt:704)
        at prog8.compiler.target.cpu6502.codegen.AsmGen.block2asm(AsmGen.kt:213)
        at prog8.compiler.target.cpu6502.codegen.AsmGen.compileToAssembly(AsmGen.kt:63)
        at prog8.compiler.CompilerKt.writeAssembly(Compiler.kt:343)
        at prog8.compiler.CompilerKt.compileProgram(Compiler.kt:102)
        at prog8.CompilerMainKt.compileMain(CompilerMain.kt:114)
        at prog8.CompilerMainKt.main(CompilerMain.kt:23)
`